extensions/rich_text_components/components.py

```diff
@@ -0,0 +1,172 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Classes for Rich Text Components in Oppia."""
+
+import re
+
+import bs4
+from extensions.objects.models import objects
+
+
+class BaseRteComponent(object):
+    """Base object class.
+
+    This is the superclass for rich text components in Oppia, such as
+    Image and Video.
+    """
+
+    # These values should be overridden in subclasses.
+    customization_args = {}
+
+    arg_schema = {}
+
+    @classmethod
+    def validate(cls, value_dict):
+        """Validates customization args for a rich text component.
+
+        Raises:
+          TypeError: if any customization arg is invalid.
+        """
+        if cls.arg_schema:
+            required_attrs = cls.arg_schema
+        else:
+            required_attrs = cls.customization_args
+        required_attr_names = required_attrs.keys()
+        attr_names = value_dict.keys()
+        if set(attr_names) != set(required_attr_names):
+            raise Exception('Invalid attributes')
+
+        for arg_name in required_attr_names:
+            arg_obj_class = required_attrs[arg_name]
+            arg_obj_class.normalize(value_dict[arg_name])
+
+
+class Collapsible(BaseRteComponent):
+    """Class for Collapsible component."""
+
+    customization_args = {
```

Comment:
Have backend tests to ensure these schemas match those in rich_text_components_definitions.js. Or use that JS file as a single source of truth for the schemas instead.

Reply:
Used the js file to find customization args.